include/util/query_heap.hpp

```diff
@@ -180,7 +180,9 @@ class QueryHeap
     {
         BOOST_ASSERT(WasInserted(node));
         const Key index = node_index.peek_index(node);
-        return inserted_nodes[index].handle == HeapHandle{};
+        auto const end_it = const_cast<HeapContainer&>(heap).end(); // non-const iterator
```

Comment:
interestingly this method is used only in `DecreaseKey` in assertion. 
I guess  it can be removed here and in the tests,
and in `DecreaseKey` 
`BOOST_ASSERT(reference.handle != HeapContainer::s_handle_from_iterator(heap.end())); // was not removed`

Reply:
@daniel-j-h & @TheMarex shall I update the PR as you suggest or simply remove the method completely as @oxidase suggests?

Reply:
@mloskot I would like to keep the interface change out of this PR. :-)

Reply:
@TheMarex Makes sense.